project/lit/views.py

```diff
@@ -38,6 +39,49 @@ def get_context_data(self, **kwargs):
         context['tags'] = models.ReferenceFilterTag.get_all_tags(self.assessment.id)
         return context
 
+class BulkTagging(TemplateView):
+    template_name = "lit/bulk_tagging.html"
+
+    def get_context_data(self, **kwargs):
+        context = super().get_context_data(**kwargs)
+        self.assessment = get_object_or_404(Assessment, pk=kwargs['pk'])
+        context['assessment'] = self.assessment
+        context['tags'] = models.ReferenceFilterTag.get_all_tags(self.assessment.id)
+        context['method'] = 'GET'
+        return context
+
+    def post(self, request, *args, **kwargs):
+        context = self.get_context_data(**kwargs)
+        context['method'] = 'POST'
+
+        ref_ids = (self.request.POST["referenceIds"]).split(",")
+        tag_ids = (self.request.POST["tagIds"]).split(",")
+
+        self.process_bulk_tag_request(ref_ids, tag_ids)
+
+        return super(TemplateView, self).render_to_response(context)
+
+    @atomic
+    def process_bulk_tag_request(self, reference_ids_strings, tag_ids_strings):
```

Comment:
done, added this as a static method in the models.Reference class. I hadn't seen those type/return hints before, interesting. This will be in next commit.